### PR TITLE
[SPARK-47614][CORE][DOC] Update some outdated comments about `JavaModuleOptions`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to update some outdated comments about `JavaModuleOptions`.

### Why are the changes needed?
As more and more options for JVM runtime are added to the class `JavaModuleOptions`, some doc comments about `JavaModuleOptions` have outdated.
Let's update it.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
N/A.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @LuciferYang @dongjoon-hyun

**Review by dongjoon-hyun — COMMENTED** [review]
Although I understand the rational, I'm not sure about this at this time.

**Comment by LuciferYang** [comment]
```
[error] /home/runner/work/spark/spark/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:61:8: object JavaModuleOptions is not a member of package org.apache.spark.launcher
[error] import org.apache.spark.launcher.{JavaModuleOptions, LauncherBackend, SparkAppHandle, YarnCommandBuilderUtils}
[error]        ^
[error] /home/runner/work/spark/spark/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:[1043](https://github.com/panbingkun/spark/actions/runs/8448161016/job/23140070843#step:10:1044):17: not found: value JavaModuleOptions
[error]     javaOpts += JavaModuleOptions.defaultModuleOptions()
[error]                 ^
[error] /home/runner/work/spark/spark/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:61:35: Unused import
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=unused-imports, site=org.apache.spark.deploy.yarn
[error] import org.apache.spark.launcher.{JavaModuleOptions, LauncherBackend, SparkAppHandle, YarnCommandBuilderUtils}
[error]                                   ^
[warn] one warning found
[error] three errors found
```

**Comment by LuciferYang** [comment]
I don't oppose this modification. However, due to the inclusion of an option other than `--add-opens`, it does deviate from the original design intent. But as a `public` class, renaming it should indeed be approached with caution.

also cc @HyukjinKwon @cloud-fan @yaooqinn @wangyum

**Comment by panbingkun** [comment]
> ```
> 
> [error] /home/runner/work/spark/spark/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:61:8: object JavaModuleOptions is not a member of package org.apache.spark.launcher
> 
> [error] import org.apache.spark.launcher.{JavaModuleOptions, LauncherBackend, SparkAppHandle, YarnCommandBuilderUtils}
> 
> [error]        ^
> 
> [error] /home/runner/work/spark/spark/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:[1043](https://github.com/panbingkun/spark/actions/runs/8448161016/job/23140070843#step:10:1044):17: not found: value JavaModuleOptions
> 
> [error]     javaOpts += JavaModuleOptions.defaultModuleOptions()
> 
> [error]                 ^
> 
> [error] /home/runner/work/spark/spark/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala:61:35: Unused import
> 
> [error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=unused-imports, site=org.apache.spark.deploy.yarn
> 
> [error] import org.apache.spark.launcher.{JavaModuleOptions, LauncherBackend, SparkAppHandle, YarnCommandBuilderUtils}
> 
> [error]                                   ^
> 
> [warn] one warning found
> 
> [error] three errors found
> 
> ```

Let me to update it.

**Comment by yaooqinn** [comment]
I prefer to keep it AS-IS rather than changing it eagerly, especially things with `since` tags

File: launcher/src/main/java/org/apache/spark/launcher/JVMRuntimeOptions.java [hunk 1]
```diff
@@ -18,14 +18,14 @@
 package org.apache.spark.launcher;
 
 /**
- * This helper class is used to place the all `--add-opens` options
- * required by Spark when using Java 17. `DEFAULT_MODULE_OPTIONS` has added
+ * This helper class is used to place some JVM runtime options(eg: `--add-opens`)
+ * required by Spark when using Java 17. `DEFAULT_OPTIONS` has added
  * `-XX:+IgnoreUnrecognizedVMOptions` to be robust.
  *
  * @since 3.3.0
```
Comment (cloud-fan) [inline launcher/src/main/java/org/apache/spark/launcher/JVMRuntimeOptions.java:25]: If this is a public API, let's not touch it. We can correct the doc comments though.

Comment (panbingkun) [reply launcher/src/main/java/org/apache/spark/launcher/JVMRuntimeOptions.java:25]: Okay, let me to correct the doc comments in this pr.

**Review by cloud-fan — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thanks @panbingkun @LuciferYang @dongjoon-hyun @cloud-fan 

Merged to master.

Hi, @panbingkun please also update the Jira ticket name, thank you

**Comment by panbingkun** [comment]
> Thanks @panbingkun @LuciferYang @dongjoon-hyun @cloud-fan
> 
> Merged to master.
> 
> Hi, @panbingkun please also update the Jira ticket name, thank you

Okay, let me to update it.

**Pull request closed without merging**